### PR TITLE
dev-scripts bump

### PR DESCRIPTION
**Pull request by iowillhoit**
### What does this PR do?
Bumps dev-scripts and cleans up a few things

- Dev-scripts bump changes a lot of dev deps. Minor changes needed to pass tests
- Dev-scripts bump adds `link-check`. Updates `plugin-marketplace` to fix broken link
- Ran `oclif readme` (to pass `link-check`)
  - Adds a `readme` script with the [flags we used in PRM](https://github.com/salesforcecli/plugin-release-management/blob/4de29e54def36cc73a486bc0527a555caa7302f0/src/commands/cli/release/build.ts#L194) 
- Sha of `ncipollo/release-action` used instead of Release version

[skip-validate-pr]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -238,41 +239,22 @@
...
-    "@salesforce/prettier-config": "^0.0.3",
     "@salesforce/ts-sinon": "^1.4.19",
     "@types/debug": "^4.1.12",
-    "@typescript-eslint/eslint-plugin": "^5.62.0",
-    "@typescript-eslint/parser": "^5.62.0",
     "aws-sdk": "^2.1511.0",
-    "chai": "^4.3.10",
-    "eslint": "^8.55.0",
     "eslint-config-oclif": "^5",
     "eslint-config-oclif-typescript": "^2.0.1",
```
Comment (mshanemc) [inline package.json:254]: is there really anything in here that we need?

File: package.json [hunk 2]
```diff
@@ -345,6 +327,15 @@
         ".nycrc"
       ],
       "output": []
+    },
+    "link-check": {
```
Comment (mshanemc) [inline package.json:331]: it probably doesn't hurt, but it won't run often since humans don't do much with the top-level cli

**Review by mshanemc — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (iowillhoit) [reply package.json:254]: Hm, i don't know why we would need these. @mdonnalley do you know?

File: README.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ $ npm install -g @salesforce/cli
 $ sf COMMAND
 running command...
 $ sf (--version|-v)
-@salesforce/cli/2.23.0 linux-x64 node-v20.10.0
+@salesforce/cli/2.23.0 darwin-arm64 node-v20.10.0
```
Comment (iowillhoit) [inline README.md:27]: This will revert when the next `nightly` builds in CI

**Pull request merged**